src/Core/AndroidNative/maui/src/main/java/com/microsoft/maui/PlatformInterop.java

```diff
@@ -455,6 +461,53 @@ private static boolean isColorType(TypedValue value)
         }
     }
 
+    /**
+     * Sets the maxLength of an EditText
+     * @param editText
+     * @param maxLength
+     */
+    public static void updateMaxLength(@NonNull EditText editText, int maxLength)
+    {
+        setLengthFilter(editText, maxLength);
+
+        Editable currentText = editText.getText();
+        if (currentText.length() > maxLength)
+        {
+            editText.setText(currentText.subSequence(0, maxLength));
+        }
+    }
+
+    /**
+     * Updates the InputFilter[] of an EditText. Used for Entry and SearchBar.
+     * @param editText
+     * @param maxLength
+     */
+    public static void setLengthFilter(@NonNull EditText editText, int maxLength)
+    {
+        if (maxLength == -1)
+            maxLength = Integer.MAX_VALUE;
+
+        List<InputFilter> currentFilters = new ArrayList<>(Arrays.asList(editText.getFilters()));
```

Comment:
C# had a null check for `getFilters()` (`?? new IInputFilters[0]`), but I don't think it can return null:

https://github.com/aosp-mirror/platform_frameworks_base/blob/da78d227b2b5ae6faa2b892c7863531f1b018985/core/java/android/text/SpannableStringBuilder.java#L1595-L1607

I can't find any null checks for it in Android's source code:

https://github.com/search?q=repo%3Aaosp-mirror%2Fplatform_frameworks_base%20getFilters&type=code